### PR TITLE
SI-7497 Fix scala.util.Properties.isMac

### PR DESCRIPTION
**Pull request by soc**
It returned wrong values because it used the java.vendor
property instead of os.name, as specified in
developer.apple.com/library/mac/#technotes/tn2002/tn2110.

### CODE REVIEW

**Comment by soc** [comment]
Review by @phaller or @paulp (because I remember that he uses Mac OSX), please.

File: src/library/scala/util/Properties.scala [hunk 1]
```diff
@@ -136,10 +135,11 @@ private[scala] trait PropertiesTrait {
...
   def userName              = propOrEmpty("user.name")
 
-  /** Some derived values.
-   */
+  /* Some derived values. */
+  /** Returns `true` iff the underlying operating system is a version of Microsoft Windows. */
   def isWin                 = osName startsWith "Windows"
-  def isMac                 = javaVendor startsWith "Apple"
+  /** Returns `true` iff the underlying operating system is a version of Apple Mac OSX.  */
+  def isMac                 = osName contains   "OS X" // See developer.apple.com/library/mac/#technotes/tn2002/tn2110
```
Comment (paulp) [inline src/library/scala/util/Properties.scala]: This is kind of an obnoxious recommendation on Apple's part. I know if I were one of the developers of "BootOS Xanadu" I wouldn't much appreciate it.

Comment (adriaanm) [inline src/library/scala/util/Properties.scala]: [apache commons](http://commons.apache.org/proper/commons-lang/javadocs/api-3.1/src-html/org/apache/commons/lang3/SystemUtils.html#line.1426) uses `startsWith("Mac OS X")`

Comment (paulp) [inline src/library/scala/util/Properties.scala]: More than you ever wanted to know: http://mail.openjdk.java.net/pipermail/macosx-port-dev/2012-November/005148.html

My summary: do what adriaan said.

Comment (soc) [inline src/library/scala/util/Properties.scala]: Ok, too bad that the backport was already merged ... I really wonder why I end up having PRs in a half-merged state.

**Comment by soc** [comment]
@paulp, @adriaanm: Updated.

**Comment by soc** [comment]
Backport: https://github.com/scala/scala/pull/2587

**Comment by adriaanm** [comment]
LGTM

**Pull request merged**